src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1226,7 +1242,7 @@ CATCH_RETURN();
     {
         // Get the font fallback first
         ::Microsoft::WRL::ComPtr<IDWriteTextFormat1> format1;
-        if (FAILED(_format.As(&format1)))
+        if (FAILED(::Microsoft::WRL::ComPtr<IDWriteTextFormat>(_formatInUse).As(&format1)))

```

Comment:
I thought this strange, then I realized you're storing the `InUse` ones as a bare pointer, instead of as a `ComPtr`. Given the `ComPtr` is just a refcount anyway... should we maybe just store it that way so there are 2 refs outstanding for holding it twice in our class? I don't think that should be THAT harmful to performance overall.

Reply:
Yeah I went back and forth with this a few times, but settled on the bare pointer in the end because it felt a bit wasteful having that extra ref counting for something that's entirely internal to the class. I wouldn't object to going back to a `ComPtr` if that's what you prefer though. Just let me know.

Reply:
The lifetime is held by other member variables. I am comfortable with a bare pointer here.

Anyway, it will be instantly promoted to an owning reference (of TextFormat1) here as needed.

Reply:
If you want to make this a little "clearer" (clearer to a COM person; instead of using a constructor call with `_formatInUse`), you can do this:

```
if (FAILED(_formatInUse->QueryInterface(IID_PPV_ARGS(&format1))))
```

`IID_PPV_ARGS` is a macro that takes a pointer to a COM thing and transforms it into an interface ID for that pointed-to type + the type as an argument pack.

the `&` operator on ComPtr acts roughly as expected of COM things.

QueryInterface returns a positive reference counted object into the pointed-to location so it's consistent with what WRL needs.

Reply:
OK I'm good with what Dustin says then. Bare pointer is fine with me and you can optionally make it "clearer to a COM person" per his suggestion.